### ISSUE TITLE
Reciprocal axes

### ISSUE THREAD

**Issue by buerviper** [veusz/veusz#368]
For some plots, it would be good to have a reciprocal display of an axis (e.g., one could show 2 x axes, one with an Energy scale and with wavelength - the one of them would need to be linear and the other one reciprocal).
Would it be possible to include such an option?

**Comment by jeremysanders** [comment]
Hi - Veusz can do something like this with the axis-function widget. Have a look at the axis_function_linked.vsz example.

I agree it would be nice to have more axis types in the standard axis widget, however.

**Comment by buerviper** [comment]
Oh, nice, this is at least a work-around for two axes where one of them should be reciprocal.

Sometimes, I just want to plot one axis reciprocally (for example in an non-linearized Arrhenius plot, the temperature axis would need to be reciprocal). How could I realize that with the current options?

**Comment by mythsmith** [comment]
> Hi - Veusz can do something like this with the axis-function widget. Have a look at the axis_function_linked.vsz example.
> 
> I agree it would be nice to have more axis types in the standard axis widget, however.

I could not make this work. If I specify  function=1/t, I got the error: 
Error in axis-function (1/t): 'Not a monotonic function'

**Comment by buerviper** [comment]
> 
> 
> > Hi - Veusz can do something like this with the axis-function widget. Have a look at the axis_function_linked.vsz example.
> > I agree it would be nice to have more axis types in the standard axis widget, however.
> 
> I could not make this work. If I specify function=1/t, I got the error:
> Error in axis-function (1/t): 'Not a monotonic function'

In an example I tried now, I had to specify a min/max range and linked it to my other (linear) x axis and it worked.

**Comment by joanimato** [comment]
@buerviper Can you post your file? I'm trying to do the same, but the function-axis will not display any values, even though I have set the min/max range. I keep getting `Warning: values exceed bounds in axis-function`

**Comment by jeremysanders** [comment]
@joanimato This is a very old bug, but attached is an example. You need to set the min and max t so that the function is monotonic. Note - rename this file to .vsz before you can use it.
[example_inv.vsz.txt](https://github.com/veusz/veusz/files/9994982/example_inv.vsz.txt)

**Comment by joanimato** [comment]
> @joanimato This is a very old bug, but attached is an example. You need to set the min and max t so that the function is monotonic. Note - rename this file to .vsz before you can use it.
[example_inv.vsz.txt](https://github.com/veusz/veusz/files/9994982/example_inv.vsz.txt)

Thanks @jeremysanders for taking the time to reply to an old thread. However, it's not exactly the example I have in mind. What I (and I think op) was trying to do is to have two axes, one linear and one reciprocal. For example, normal x-axis goes from 0 to 1 while reciprocal axis is 1/x of the x-axis. In other words, use the "linked" option. 

Setting the min value of `t` to be >0 (or any min/max value of `t` for that matter) does not fix this issue. The axis just displays nothing.  What fixed it for me was to also set the min value of `x` in the "normal" axis to be >0. It seems somehow that the `functional-axis` widget ignores the specified min/max of t and just reads the min/max of the normal x-axis. 

Not sure if this is a bug, or if I'm missing something here. But it would be great to be able to set the min/max of `x` and `t` independently. I.e. the x-axis can go over 0, but t=1/x cannot. 

I've attached a file of this example. 

[test-axis.vsz.txt](https://github.com/veusz/veusz/files/9996010/test-axis.vsz.txt)

**Comment by buerviper** [comment]
> For example, normal x-axis goes from 0 to 1 while reciprocal axis is 1/x of the x-axis. In other words, use the "linked" option.

I don't fully get this. If your original x axis starts at 0, you cannot have a reciprocal linked axis - there cannot be a value for "1/0".

In your file: change the "min t" of the linked axis to any value equal or smaller to the lowest x axis value, then it shows no error message and the axis is shown correctly.

The "issue" here is completely fixed for me.

**Comment by joanimato** [comment]
@buerviper In some cases it makes sense. For instance, if you want to extrapolate the values of a dataset to infinity, it would be beneficial to have the "0" point of x-axis in order to show where the extrapolation crosses the axis. To this end, it would be beneficial to have the functional 2nd axis linked with the x-axis, but only compute values for the given subset of x, specified by min/max t. 

The example I provided works fine, as long as x-min is > 0. but it doens't if you  set x-min to 0, which is what I would like.

**Comment by buerviper** [comment]
Ah, no I understand what you want - this is actually different from my initial question.

So you would like to have a linked axis which only displays the data range indicated by min and max t (and for "Auto", it should take the values of the linked x axis)?

In that regard, the "min position" and "max position" of the linked axis do not seem to influence anything in the graph.

**Comment by joanimato** [comment]
Yes, or you simply specify a range of x (x=t) for which the linked axis is computed. The "linked" part is only necessary so that the respective ticks of the top axis match with the bottom.
